src/browser/fetch_network.js

```diff
@@ -0,0 +1,1103 @@
+            offset++;
+            for ( let ii = 0; ii < s.length; ++ii) {
+                view.setUint8(offset, s.charCodeAt(ii));
+                offset++;
+            }
+        }
+        view.setUint16(offset, q.type);
+        offset += 2;
+        view.setUint16(offset, q.class);
+        offset += 2;
+    }
+
+    function write_reply(a) {
+        for (let s of a.name) {
+            view.setUint8(offset, s.length);
+            offset++;
+            for ( let ii = 0; ii < s.length; ++ii) {
+                view.setUint8(offset, s.charCodeAt(ii));
+                offset++;
+            }
+        }
+        view.setUint16(offset, a.type);
+        offset += 2;
+        view.setUint16(offset, a.class);
+        offset += 2;
+        view.setUint32(offset, a.ttl);
+        offset += 4;
+        view.setUint16(offset, a.data.length);
+        offset += 2;
+
+        for (let ii = 0; ii < a.data.length; ++ii) {
+            view.setUint8(offset + ii, a.data[ii]);
+        }
+
+        offset += a.data.length;
+    }
+
+    for (let i = 0; i < spec.dns.answers.length; ++i) {
+        let a = spec.dns.answers[i];
+        write_reply(a);
+    }
+
+    return offset;
+}
+
+function parse_dhcp(data, o) {
+    let view = new DataView(data.buffer, data.byteOffset, data.byteLength);
+    let bootpo  = data.subarray(44,44+192);
+    let dhcp = {
+        op:  view.getUint8(0),
+        htype: view.getUint8(1),
+        hlen: view.getUint8(2),
+        hops: view.getUint8(3),
+        xid: view.getUint32(4),
+        secs: view.getUint16(8),
+        flags: view.getUint16(10),
+        ciaddr: view.getUint32(12),
+        yiaddr: view.getUint32(16),
+        siaddr: view.getUint32(20),
+        giaddr: view.getUint32(24),
+        chaddr: data.subarray(28,28+16),
+        magic: view.getUint32(236),
+        options: [],
+    };
+
+    let options = data.subarray(240);
+    for (let i = 0; i < options.length; ++i) {
+        let start = i;
+        let op = options[i];
+        if (op == 0) continue;
+        ++i;
+        let len = options[i];
+        i += len;
+        dhcp.options.push(options.subarray(start, start + len + 2));
+    }
+
+    o.dhcp = dhcp;
+    o.dhcp_options = dhcp.options;
+    return true;
+}
+
+function write_dhcp(spec, data) {
+    let view = new DataView(data.buffer, data.byteOffset, data.byteLength);
+
+    view.setUint8(0, spec.dhcp.op);
+    view.setUint8(1, spec.dhcp.htype);
+    view.setUint8(2, spec.dhcp.hlen);
+    view.setUint8(3, spec.dhcp.hops);
+    view.setUint32(4, spec.dhcp.xid);
+    view.setUint16(8, spec.dhcp.secs);
+    view.setUint16(10, spec.dhcp.flags);
+    view.setUint32(12, spec.dhcp.ciaddr);
+    view.setUint32(16, spec.dhcp.yiaddr);
+    view.setUint32(20, spec.dhcp.siaddr);
+    view.setUint32(24, spec.dhcp.giaddr);
+
+    for (let i = 0; i < spec.dhcp.chaddr.length; ++i) {
+        view.setUint8(28+i, spec.dhcp.chaddr[i]);
+    }
+
+    view.setUint32(236, 0x63825363);
```

Comment:
How about idea to replace it to:
```js
view.setUint32(236, DHCP_MAGIC_COOKIE);
```
and set in upper:
```js
const DHCP_MAGIC_COOKIE = 0x63825363;
```
as small improvement of readability code (like `IPV4_PROTO_*` and `ETHERTYPE_*` variables)?